debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PGTableSchemaBuilder.java

```diff
@@ -195,7 +195,7 @@ protected StructGenerator createKeyGenerator(Schema schema, TableId columnSetNam
             if (connectorConfig.plugin().isYBOutput()) {
               value = converter.convert(((Object[]) value)[0]);
             } else {
-              value = converter.convert(value);
+              throw new DebeziumException("Class not supposed to be used with the plugin " + connectorConfig.plugin().getPostgresPluginName() + ", check configuration");
```

Comment:
Why are we throwing here? It should be done at the higher level like constructor or creating TableSchema